pkg/handler/handler.go

```diff
@@ -18,9 +20,33 @@ func Handle(payload interface{}) error {
 	switch payload.(type) {
 
 	case github.PullRequestPayload:
-		return pullrequest.Handle(payload.(github.PullRequestPayload))
+		pr := payload.(github.PullRequestPayload)
+		mutex := getMutex(pr.PullRequest.Base.Repo.FullName)
+		mutex.Lock()
+		defer mutex.Unlock()
+		return pullrequest.Handle(pr)
+
 	case github.PullRequestReviewPayload:
-		return handlePullRequestReview(payload.(github.PullRequestReviewPayload))
+		prReview := payload.(github.PullRequestReviewPayload)
```

Comment:
This is probably cleaner if I move it all to the git module, @mkolesnik what do you think?